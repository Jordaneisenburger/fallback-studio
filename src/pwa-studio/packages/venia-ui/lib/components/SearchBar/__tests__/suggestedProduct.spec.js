import React from 'react';
import TestRenderer from 'react-test-renderer';

import SuggestedProduct from '../suggestedProduct';

jest.mock('../../../classify');
jest.mock('@magento/venia-drivers', () => ({
    Link: ({ children }) => children,
    resourceUrl: jest.fn(src => src)
}));

const defaultProps = {
    handleOnProductOpen: jest.fn(),
    url_key: 'urlKey',
    small_image: '/media/catalog/category/minimalist.jpg',
    name: 'Product Name',
    price: {
        regularPrice: {
            amount: {
                currency: 'USD',
                value: 3.5
            }
        }
    }
};

test('renders a suggestedProduct component', () => {
    const component = TestRenderer.create(
        <SuggestedProduct {...defaultProps} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});
