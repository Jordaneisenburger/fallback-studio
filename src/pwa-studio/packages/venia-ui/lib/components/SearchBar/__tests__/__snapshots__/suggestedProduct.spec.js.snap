// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a suggestedProduct component 1`] = `
Array [
  <span
    className="image"
  >
    <img
      alt="Product Name"
      className="thumbnail"
      sizes="60px"
      src="/media/catalog/category/minimalist.jpg"
      srcSet="/media/catalog/category/minimalist.jpg 40w,/media/catalog/category/minimalist.jpg 80w,/media/catalog/category/minimalist.jpg 160w,/media/catalog/category/minimalist.jpg 320w,/media/catalog/category/minimalist.jpg 640w,/media/catalog/category/minimalist.jpg 1280w,/media/catalog/category/minimalist.jpg 1600w,/media/catalog/category/minimalist.jpg 2560w"
    />
  </span>,
  <span
    className="name"
  >
    Product Name
  </span>,
  <span
    className="price"
  >
    <div />
  </span>,
]
`;
